parquet-avro/src/test/java/parquet/avro/TestAvroReadSupport.java

```diff
@@ -0,0 +1,198 @@
+package parquet.avro;
+
+import java.util.*;
+import org.apache.avro.Schema;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import parquet.hadoop.api.InitContext;
+import parquet.schema.MessageType;
+import parquet.schema.MessageTypeParser;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static parquet.avro.AvroReadSupport.*;
+
+public class TestAvroReadSupport {
```

Comment:
These are great tests. It would be good to have one (in TestReadWrite?) that additionally checks that records can actually be read correctly when there are different writer schemas.